File: pippip-native-ios/API/ClientAuthorized.m
//
//  ClientAuthorized.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 12/6/17.
//  Copyright © 2017 seComm. All rights reserved.
//

#import "ClientAuthorized.h"
#import "pippip_native_ios-Swift.h"

@interface ClientAuthorized ()
{
    
    SessionState *sessionState;
    
}
@end

@implementation ClientAuthorized

- (instancetype)init {
    self = [super init];
    
    sessionState = [[SessionState alloc] init];
    return self;
    
}

- (BOOL)processResponse:(NSDictionary *)response errorDelegate:(id<ErrorDelegate>)errorDelegate {
    
    NSString *errorStr = [response objectForKey:@"error"];
    NSNumber *authToken = [response objectForKey:@"authToken"];
    sessionState.authToken = [authToken longLongValue];
    if (errorStr != nil) {
        [errorDelegate responseError:errorStr];
        return NO;
    }

    return YES;

}

@end

File: pippip-native-ios/API/ClientAuthorized.swift
//
//  ClientAuthorized.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 5/29/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit
import ObjectMapper

class ClientAuthorized: NSObject, APIResponseProtocol {

    var error: String?
    var sessionId: Int32?
    var authToken: Int64?

    var alertPresenter = AlertPresenter()
    var sessionState = SessionState()

    required init?(map: Map) {
        if map.JSON["sessionId"] == nil {
            return nil
        }
        if map.JSON["authToken"] == nil {
            return nil
        }
    }
    
    func mapping(map: Map) {

        sessionId <- map["sessionId"]
        error <- map["error"]
        authToken <- map["authToken"]

    }
    
    func processResponse() throws {

        if let responseError = error {
            throw APIResponseError.serverResponseError(error: responseError)
        }

        sessionState.authToken = authToken!

    }
    

}
